src/ResourceManager/ServerManagement/Commands.ServerManagement.Test/ScenarioTests/ServerManagementTests.ps1

```diff
@@ -0,0 +1,249 @@
+$recording = [Microsoft.Azure.Commands.ServerManagement.Test.ScenarioTests.ServerManagementTests]::Recording
+$installed = [Microsoft.Azure.Commands.ServerManagement.Test.ScenarioTests.ServerManagementTests]::GatewayServiceInstalled 
+$admin = [Microsoft.Azure.Commands.ServerManagement.Test.ScenarioTests.ServerManagementTests]::IsAdmin 
+$nodeName= [Microsoft.Azure.Commands.ServerManagement.Test.ScenarioTests.ServerManagementTests]::NodeName 
+$nodePassword = [Microsoft.Azure.Commands.ServerManagement.Test.ScenarioTests.ServerManagementTests]::NodePassword
+$nodeUserName = [Microsoft.Azure.Commands.ServerManagement.Test.ScenarioTests.ServerManagementTests]::NodeUserName
+$sessionId= [Microsoft.Azure.Commands.ServerManagement.Test.ScenarioTests.ServerManagementTests]::SessionId
+
+$creds = New-Object System.Management.Automation.PSCredential -ArgumentList $nodeUserName, (ConvertTo-SecureString $nodePassword -AsPlainText -Force)
+
+function Ensure-GatewayIsCorrectlyConfigured {
+	$GatewayName = "mygateway"
+
+	if ($recording)  {
+		# gateway service must be installed on recording machine for this stuff to work.
+
+		# check to see if the gateway service is actually installed.
+		Assert-True { $installed } "Gateway Service Is Not Installed on this computer."
+
+		# and that we are actually admin
+		Assert-True { $admin } "Recording for this test requires an elevated process."
+
+		# and let's make sure that the gateway is created in the RP
+		Write-Verbose "Getting gateway $gatewayName"
+		$gateway = Get-AzureRmServerManagementGateway -ResourceGroupName $resourceGroupName -GatewayName $GatewayName -ea 0
+		
+		if( -not $gateway ) {
+			# Well, don't panic, we can register the gateway and download and install the profile.
+			Write-Verbose "Gateway $gatewayName is not registered in the RP; we're going to register it."
+			
+
+			Write-Verbose "Stopping Gateway Service."
+			$null = stop-service ServerManagementToolsGateway -ea 0
+
+			Write-Verbose "Creating Gateway in RP"
+			$gateway = new-AzureRmServerManagementGateway -resourceGroupName $resourceGroupName -Location $location -gatewayname $GatewayName
+
+			Write-Verbose "Resetting Profile for gateway $gatewayName"
+			Reset-AzureRmServerManagementGatewayProfile $gateway 
+
+			Write-Verbose "Downloading Profile for gateway $gatewayName"
+			Save-AzureRmServerManagementGatewayProfile  $gateway -outputfile "$env:TMP\gatewayprofile.json"
+
+			Write-Verbose "Checking for profile"
+			assert-true { Test-Path "$env:TMP\gatewayprofile.json" } "Profile Does Not Exist!"
+
+			Write-Verbose "Installing profile"
+			Install-AzureRmServerManagementGatewayProfile  "$env:TMP\gatewayprofile.json" 
+			
+			Write-Verbose "Starting Gateway Service."
+			$null = start-service ServerManagementToolsGateway -ea 0
+
+			Write-Verbose "Waiting 3 minutes for Gateway Service to be in the correct state."
+			sleep 180
```

Comment:
There is a check to see if it's recording above. the whole section is excluded if it's not recording.